test/src/test/java/org/corfudb/runtime/object/transactions/OptimisticTransactionContextTest.java

```diff
@@ -109,6 +111,50 @@ public void customConflictObjectsConflictAborts()
                     .isInstanceOf(TransactionAbortedException.class);
     }
 
+    /**
+     * This test checks if optimistic transactions are aborted in the scenario of slow writers and
+     * fast readers accessing the same stream, i.e., a reader accesses the address already given to
+     * the slow writer before this one gets to actually write into it. This causes the address to
+     * be hole filled and therefore the transaction is aborted after several tries.
+     * The cause of abort of this transaction is reported as OVERWRITE given that the position
+     * was continuously overwritten at the log unit level.
+     */
+    @Test
+    public void checkSlowWriterAbortsOnTransaction() {
+        UUID streamID = UUID.randomUUID();
+        Boolean caughtException = false;
+        CorfuRuntime rtWriter = getDefaultRuntime();
+        CorfuRuntime rtReader = getDefaultRuntime();
+
+        Map<String, String> map = rtWriter
+                .getObjectsView().build()
+                .setStreamID(streamID)
+                .setType(SMRMap.class)
+                .open();
+
+        try {
+            OptimisticTXBegin();
+            map.put("k1", "v1");
+            TestRule testRule = new TestRule()
+                    .matches(m -> {
+                        if (m.getMsgType().equals(CorfuMsgType.WRITE)) {
+                            rtReader.getStreamsView().get(streamID).next();
+                            return true;
+                        } else {
+                            return false;
+                        }
+                    });
+            addClientRule(rtWriter, testRule);
+            TXEnd();
```

Comment:
Add a fail after TXEnd. `Assert.fail()` So that the test fails if the transaction completes without aborting.
You can then get rid of your caughtException flag.

Reply:
Done